lib/p2p/Peer.ts

```diff
@@ -173,19 +177,23 @@ class Peer extends EventEmitter {
 
     await this.initConnection(retryConnecting);
     this.initStall();
-    await this.initHello(handshakeData);
+    await this.initHello(handshakeState);
 
     if (this.expectedNodePubKey && this.nodePubKey !== this.expectedNodePubKey) {
       this.close(DisconnectionReason.UnexpectedIdentity);
       throw errors.UNEXPECTED_NODE_PUB_KEY(this.nodePubKey!, this.expectedNodePubKey, addressUtils.toString(this.address));
     }
 
-    if (this.nodePubKey === handshakeData.nodePubKey) {
+    if (this.nodePubKey === handshakeState.nodePubKey) {
       this.close(DisconnectionReason.ConnectedToSelf);
       throw errors.ATTEMPTED_CONNECTION_TO_SELF;
     }
 
-    // TODO: Check that the peer's version is compatible with ours
+    // Check version compatibility
+    if (this.version !== handshakeState.version) {
```

Comment:
not every version update will be non-compatible. We need to define `MIN_VERSION`, and verify the compatibility by converting the string to numbers (or something similar to that).

Reply:
As Daniel said in https://github.com/ExchangeUnion/xud/issues/154#issuecomment-436832861  we need a separate field indicating a min compatible version. We will set this manually with every release. The `MIN_VERSION` will be = current version if the current version introduces a significant breaking change e.g. on p2p level. I don't see any other way apart from doing it manually on release tag.

So instead of `if (this.version !== handshakeData.version)` something like `if (this.minversion !== handshakeData.version)` @rsercano

Reply:
and take care we cannot forget to set `MIN_VERSION` when tagging. Any idea about this? @sangaman

Reply:
I’d suggest to define `minCompatibleVersion` field in `package.json` and read it from there in `Xud.ts`, same as with `version`. 
We just need to upgrade it when we're upgrading `version`, before a non-compatible release.

Both the min version and the peer version strings should be parsed, and get evaluated so that the peer version needs to be higher or equal to the min version.

@kilian are we planning to go from v1.0.0-alpha.x to v1.0.0?

Reply:
When we'll get to v1.0.0, we want to increment the first number for incompatible changes release, instead of the min version field?

Reply:
@moshababo actually that was my entry point for the my first question; 
`would it be sufficient to check major version numbers ?`

Reply:
As a general rule per [semver](https://semver.org/), once we are at v1.0.0 we'll increment MAJOR in vMAJOR.MINOR.PATCH for every breaking change that effects the majority of users. Backwards-compatible or breaking changes for a small subset of users might go via MINOR, we'll use our judgement on a case by case base.

Nevertheless this is just for the sake of readability, your check shouldn't care about this and just go for the `minCompatibleVersion` field @rsercano 

Btw, I got zero feeling for staying in low version numbers as long as possible (like bitcoin) a lot of major software products nowadays are in double digits which is perfectly fine. But of course we should reduce breaking changes to a minimum once we are at v1.0.0.

Reply:
Alright, I've implemented semantic version check which has a logic as below;

- 1.0.0 < 2.0.0 < 2.1.0 < 2.1.1
- 1.0.0-alpha < 1.0.0-alpha.1 < 1.0.0-alpha.beta < 1.0.0-beta < 1.0.0-beta.2 < 1.0.0-beta.11 < 1.0.0-rc.1 < 1.0.0

Reply:
Looks right to me.